### PR TITLE
v18.2.4

### PR DESCRIPTION
**Pull request by yuriw**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: PendingReleaseNotes [hunk 1]
```diff
@@ -100,6 +100,13 @@
    `ceph config get mgr mgr/volumes/snapshot_clone_no_wait`
   and it can be disabled by using:
    `ceph config set mgr mgr/volumes/snapshot_clone_no_wait false`
+* CephFS: fixes to the implementation of the ``root_squash`` mechanism enabled
+  via cephx ``mds`` caps on a client credential require a new client feature
+  bit, ``client_mds_auth_caps``. Clients using credentials with ``root_squash``
+  without this feature will trigger the MDS to raise a HEALTH_ERR on the
```
Comment (anthonyeleven) [inline PendingReleaseNotes:106]: s/raise a/raise/

File: debian/control [hunk 2]
```diff
@@ -352,6 +352,30 @@ Description: debugging symbols for ceph-mgr
...
+Package: ceph-exporter
+Architecture: linux-any
+Depends: ceph-base (= ${binary:Version}),
+Description: metrics exporter for the ceph distributed storage system
+ Ceph is a massively scalable, open-source, distributed
+ storage system that runs on commodity hardware and delivers object,
+ block and file system storage.
+ .
+ This package contains the metrics exporter daemon, which is used to expose
+ the performance metrics.
```
Comment (anthonyeleven) [inline debian/control:364]: s/the performance/performance and configuration/

**Review by ljflores — APPROVED** [review]
Looks good commit-wise; I'd just apply @anthonyeleven's comments as well.

File: PendingReleaseNotes (hunk 1 above)
Comment (yuriw) [reply PendingReleaseNotes:106]: @anthonyeleven what exactly do you want to change here?

**Comment by yuriw** [comment]
@ljflores @anthonyeleven I think those commit messages don't have to be modified in this PR.  This PR is manually generated because of our build script problems and I see no point in modifying it here.
 What do you think?

File: PendingReleaseNotes (hunk 1 above)
Comment (anthonyeleven) [reply PendingReleaseNotes:106]: Removing " a " on line 106.

**Comment by anthonyeleven** [comment]
My suggestions aren't commit messages.

**Comment by yuriw** [comment]
> My suggestions aren't commit messages.

If you are referring to the release notes AFAIK   they are all fixed

https://github.com/ceph/ceph/pull/56932
https://github.com/ceph/ceph.io/pull/677

**Review by idryomov — APPROVED** [review]
> Looks good commit-wise

+1, now that https://github.com/ceph/ceph/pull/57794 is merged

**Review by guits — APPROVED** [review]
ceph-volume approval

**Comment by idryomov** [comment]
I rebased reef-release to get rid of the outdated 18.2.3 tag (from May 30, one day before centos 8 went EOL).

We would have needed a new tag anyway, and today in the CLT meeting it was decided that this release would be re-numbered to 18.2.4 due to https://download.ceph.com/tarballs/ceph_18.2.3.orig.tar.gz getting uploaded and packaged by Debian prematurely.

**Comment by idryomov** [comment]
> today in the CLT meeting it was decided that this release would be re-numbered to 18.2.4

https://tracker.ceph.com/issues/66862
https://tracker.ceph.com/issues/66863

**Comment by jdurgin** [comment]
jenkins test api

**Comment by cbodley** [comment]
reef builds seem to be broken since this merged: https://shaman.ceph.com/builds/ceph/reef/

**Comment by cbodley** [comment]
the ceph-dev-new-setup jobs are failing with:
```
++ git describe --abbrev=8 --match 'v*'
++ sed 's/^v//'
+ cephver=18.2.2-1838-g3bb8b788
+ echo current version 18.2.2-1838-g3bb8b788
current version 18.2.2-1838-g3bb8b788
++ pwd
+ srcdir=/home/jenkins-build/build/workspace/ceph-dev-setup
+ '[' -d release/18.2.2-1838-g3bb8b788 ']'
+ dch -v 18.2.2-1838-g3bb8b788-1 autobuilder
dch warning: neither DEBEMAIL nor EMAIL environment variable is set
dch warning: building email address from username and mailname
dch: fatal error at line 1222:
New version specified (18.2.2-1838-g3bb8b788-1) is less than
the current version number (18.2.4-1)!  Use -b to force.
Build step 'Execute shell' marked build as failure
```

**Pull request merged**